File: gplx-be/src/main/java/com/demo/entities/Answer.java
package com.demo.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "answer")
public class Answer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "question_id")
    private Question question;

    @Column(columnDefinition = "text")
    private String content;

    @Column(name = "is_correct")
    private boolean isCorrect;

    private boolean status;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Question getQuestion() {
        return question;
    }

    public void setQuestion(Question question) {
        this.question = question;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public boolean isCorrect() {
        return isCorrect;
    }

    public void setCorrect(boolean correct) {
        isCorrect = correct;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }
}

File: gplx-fe/lib/entities/Answer.dart
class Answer {
  int? id;
  String? content;
  bool? correct;
  bool? status;

  Answer({
    this.id,
    this.content,
    this.correct,
    this.status,
  });

  factory Answer.fromMap(Map<String, dynamic> map) {
    return Answer(
      id: map['id'] as int?,
      content: map['content'] as String?,
      correct: map['correct'] as bool?,
      status: map['status'] as bool?,
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'content': content,
      'correct': correct,
      'status': status,
    };
  }
}
